client/my-sites/themes/controller.jsx

```diff
@@ -146,17 +107,17 @@ export function redirectFilterAndType( { res, params: { site, filter, tier } } )
 	} else {
 		parts = [ tier, site ];
 	}
-	res.redirect( '/themes/' + compact( parts ).join( '/' ) );
+	res.redirect( '/themes/' + parts.filter( Boolean ).join( '/' ) );
 }
 
 export function redirectToThemeDetails( { res, params: { site, theme, section } }, next ) {
 	// Make sure we aren't matching a site -- e.g. /themes/example.wordpress.com or /themes/1234567
-	if ( includes( theme, '.' ) || isFinite( theme ) ) {
+	if ( theme.includes( '.' ) || isFinite( theme ) ) {
```

Comment:
If `theme` is a number, won't `theme.includes()` crash? Or is `theme` always a string? But if it is, then how would `isFinite( theme )` even work if it's a string?

Reply:
In my testing, when using a url with reversed, incorrect parameters like `calypso.localhost:3000/themes/twentytwentyone/179276076`, then theme is still a string (`'179276076'`).  Theme is also a string when visiting `calypso.localhost:3000/themes/179276076`

Reply:
👍  If the theme is a string in that scenario, we'll need to convert it to a number before we call `isFinite()` on it. Otherwise, `isFinite()` will always return `false`.

Reply:
We have a helper that does the "is it a site slug" check somewhere. I'll find and use it.

`context.param.theme` is a mandatory path param, always a string, always present.

Reply:
Turns out functions in `lib/route/path` do the check with `Number.isSafeInteger( parseInt( slug, 10 ) )`, but it's not extracted as a helper function. I went with `Number.isInteger( parseInt( ... ) )` which is good enough. In 0ed3958.

Reply:
Sounds good to me 👍